crates/data_components/src/odbc.rs

```diff
@@ -81,11 +88,18 @@ where
     ) -> Result<Arc<dyn TableProvider + 'static>, Box<dyn std::error::Error + Send + Sync>> {
         let pool = Arc::clone(&self.pool);
         let dyn_pool: Arc<ODBCDbConnectionPool<'a>> = pool;
-        let table_provider = Arc::new(
-            SqlTable::new("odbc", &dyn_pool, table_reference, Some(Engine::ODBC))
-                .await
-                .map_err(|e| Box::new(e) as Box<dyn std::error::Error + Send + Sync>)?,
-        );
+
+        let table = SqlTable::new("odbc", &dyn_pool, table_reference, Some(Engine::ODBC))
```

Comment:
Does the `Engine::ODBC` still make sense to have given the different dialects?

Reply:
It could still make sense, but I think the only other option we could supply here that would also make sense would be `None`.

Reply:
Right, I mean - what does this actually change, and is this really a dialect not an engine that we should be passing?

Reply:
It changes the behavior of `to_sql_with_engine` for some expressions. E.g. for columns: https://github.com/datafusion-contrib/datafusion-table-providers/blob/02081887de7b64dfd816ef86716f8b0853ae3196/src/sql/sql_provider_datafusion/expr.rs#L48

Do we use `to_sql_with_engine`? If we bypass it in favor of something else then we can probably supply `None`.

Reply:
It shouldn't be used - its a fallback in case federation isn't enabled, but we always enable federation now. At some point we should remove our own expr_to_sql implementation and just rely on the DataFusion one, in which case this gets removed anyway.